probe-rs/src/bin/probe-rs/cmd/run.rs

```diff
@@ -91,20 +97,32 @@ impl Cmd {
                 loader,
                 self.chip_erase,
             )?;
+            // reset the core to leave it in a consistent state after flashing
+            session
+                .core(0)?
+                .reset_and_halt(Duration::from_millis(100))?;
         }
 
         let memory_map = session.target().memory_map.clone();
         let rtt_scan_regions = session.target().rtt_scan_regions.clone();
         let mut core = session.core(0)?;
 
-        if run_download {
+        if self.catch_hardfault || self.catch_reset {
```

Comment:
The `reset_and_halt` should only be done if `run_download` is true, otherwise it breaks the `attach` use case. 
Right now it also seems duplicated, there is a `reset_and_halt` in line 103 as well.

Not sure how the vector catch should work tbh, I guess it should also work with attach?  Maybe just halting the core, setting the vector catch, and then running again would work.

Reply:
I've changed it to just halt when setting the vector catch, it still works in my testing but I am no means an expert in this area :D.